cli/tests/activate.bats

```diff
@@ -984,6 +984,69 @@ EOF
   assert_output --partial "test_alias is an alias for echo testing"
 }
 
+# bats test_tags=activate,activate:rc:zsh
+@test "zsh: interactive activate respects history settings from dotfile" {
+  project_setup
+
+  # This should always work, even when Darwin sets a default in `/etc/zshrc`.
+  echo 'HISTFILE=${PROJECT_DIR}/.alt_history' >"$HOME/.zshrc.extra"
+  echo 'SHELL_SESSION_DIR=${PROJECT_DIR}/.alt_sessions' >>"$HOME/.zshrc.extra"
+
+  # TODO: flox will set HOME if it doesn't match the home of the user with
+  # current euid. I'm not sure if we should change that, but for now just set
+  # USER to REAL_USER.
+  FLOX_SHELL="zsh" USER="$REAL_USER" NO_COLOR=1 \
+    run expect "$TESTS_DIR/activate/histfile.exp" "$PROJECT_DIR"
+  assert_success
+  assert_line --partial "HISTFILE=$PROJECT_DIR/.alt_history"
+  assert_line --partial "SHELL_SESSION_DIR=$PROJECT_DIR/.alt_sessions"
+}
+
+# bats test_tags=activate,activate:rc:zsh
+@test "zsh: interactive activate respects history settings from dotfile based on original ZDOTDIR" {
+  project_setup
+
+  # Mimic the default `/etc/zshrc` on Darwin prior to Nix being installed. We
+  # have to do this in `~/.zshrc` because we can't mock `/etc/zshrc`. However we
+  # apply the same `ZDOTDIR` logic to both, ensuring that it's not pointing at
+  # our immutable storepath.
+  echo 'HISTFILE=${ZDOTDIR:-$HOME}/.alt_history' >"$HOME/.zshrc.extra"
+  echo 'SHELL_SESSION_DIR=${ZDOTDIR:-$HOME}/.alt_sessions' >>"$HOME/.zshrc.extra"
+
+  # TODO: flox will set HOME if it doesn't match the home of the user with
+  # current euid. I'm not sure if we should change that, but for now just set
+  # USER to REAL_USER.
+  FLOX_SHELL="zsh" USER="$REAL_USER" NO_COLOR=1 \
+    run expect "$TESTS_DIR/activate/histfile.exp" "$PROJECT_DIR"
+  assert_success
+  assert_line --partial "HISTFILE=$HOME/.alt_history"
+  assert_line --partial "SHELL_SESSION_DIR=$HOME/.alt_sessions"
+}
+
+# bats test_tags=activate,activate:rc:zsh
+@test "zsh: interactive activate respects history settings from environment variable on Linux" {
```

Comment:
> Not sure what the mention of "on Linux" is here because it runs on Darwin too?

It was because the "respects history settings from environment variable" part wasn't true for Darwin. Will update depending on the outcome of the thread below though 👍🏻